src/main/java/org/broadinstitute/hellbender/engine/spark/JoinReadsWithVariants.java

```diff
@@ -0,0 +1,106 @@
+package org.broadinstitute.hellbender.engine.spark;
+
+import com.beust.jcommander.internal.Lists;
+import org.apache.spark.api.java.JavaPairRDD;
+import org.apache.spark.api.java.JavaRDD;
+import org.broadinstitute.hellbender.engine.dataflow.datasources.VariantShard;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import org.broadinstitute.hellbender.utils.variant.Variant;
+import scala.Tuple2;
+
+import java.util.List;
+
+/**
+ * PairReadsAndVariants takes two RDDs (GATKRead and Variant) and returns an RDD with a
+ * (GATKRead,Iterable<Variant>) for every read and all the variants that overlap. We do this by first
+ * Making KVs of GATKRead,Variant, which means that a read or variant may be present
```

Comment:
Since you're not actually using `KV` here, could you change these docs accordingly?